### ISSUE TITLE
Lost use case with `no-reserved-keys` to `quote-props` merge

### ISSUE THREAD

**Issue by freejosh** [eslint/eslint#3381]
Previously I had `quote-props` turned off and wanted errors for `no-reserved-keys` but that now seems impossible. In other words, I don't care about property quoting, except if it's a keyword.

Another way to look at it is wanting `as-needed` to have an option to ignore unnecessarily quoted properties.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by nzakas** [comment]
I'm not quite sure what you're saying. Can you show an example of what you'd like to warn and not warn?

**Comment by freejosh** [comment]
```js
/*eslint quote-props: [2, "as-needed", {"keywords": true}] */
var obj = {
	'unnecessary': 1,
	if: 2
};
```

Gives me 2 errors, but I only want the one about the reserved key and not the unnecessarily quoted property.

**Comment by ilyavolodin** [comment]
@freejosh Can you paste ESLint errors that you are getting for the code above? Also, what version of ESLint are you using and what parser?

**Comment by freejosh** [comment]
I'm using v1.1.0. For the above code I get

```
  3:5  error  Unnecessarily quoted property `unnecessary` found  quote-props
  4:5  error  Unquoted reserved word `if` used as key            quote-props
```

Which is correct for the current version, I'm just saying that I've lost the pre-1.0 behaviour where I could get the second error only.

For example, with v0.24.1 I would have used the rules `/*eslint quote-props: 0, no-reserved-keys: 2 */` to get the output:

```
  2:10  error  Reserved word 'if' used as key  no-reserved-keys
```

**Comment by nzakas** [comment]
Okay, so you only want to enforce quotes for properties if a keyword is used, and ignore all others?

**Comment by freejosh** [comment]
Yep!

**Comment by nzakas** [comment]
So maybe what we need is:

```
quote-props: [2, "ignore", { keywords: true }]
```

"ignore" is a weird term to use, but since we already have `keywords:true`, I can't think of another way.

**Comment by freejosh** [comment]
I was thinking maybe something like:

```
quote-props: [2, "as-needed", { keywords: true, unnecessary: false }]
```

To me it's still doing the `as-needed` behavior, but we're turning off a subset of results (same way as `keywords` turned on a subset)

**Comment by nzakas** [comment]
Ah! Good call, I like that.
